0.25

balloob opened the pull request:
- Allow components to register custom frontend panels (@balloob)
- New component panel_iframe allows adding other sites to sidebar (@balloob)
- Add example custom_component [`react_panel`](https://github.com/home-assistant/home-assistant/tree/dev/config/custom_components/react_panel) showing custom panels (@balloob)
- Added RFXTRX tests (@DanielHiversen)
- New IMAP sensor to monitor your inbox (@danieljkemp)
- Switch - GPIO: Fix incorrect initialization state when logic inverted (@zeroDenial)
- Telldus Live: Add support for luminance value to Fibaro Motion Sensor
- Script syntax: allow using templates for delay (@Teagan42)
- Z-Wave fixes (@turbokongen)
- input_slider will now allow float values (@ngraziano)
- Media player: Russound RNET now supported (@laf)
- Add type checking using mypy to the core (@fabianhjr)
- Thermostat: KNX based thermostat now supported (@open-homeautomation)
- Recorder stability fixes (@kellerza)
- Qwikswitch stability fixes (@kellerza)
- Thermostat - Nest: allow setting HVAC mode (@vladonemo)
- Remote classes: allow specifying custom timeout when calling Home Assistant API (@n8henrie)
- InfluxDB: option to specify additional tags (@open-homeautomation)
- Media Player: DirecTV now supported (@cbulock)
- Remote: add method to fetch config (@fabaff)
- Device Tracker - iCloud: stability fixes (@kellerza)
- Use browser timezone for frontend history and logbook dates (@armills)
- Light: X10 is now supported (@fotoetienne)
- Light - Hyperion: keep track of active color (@schneefux)
- Sensor - Speedtest: improved error handling and state restoring (@nkgilley)
- Sonos: added tests to improve code quality (@americanwookie)
- Media Player: Media Player Classic - Home Cinema now supported (@abcminiuser)
- Telegram: photo bug fixed (@pvizeli)
- Groups: lock states will now be properly grouped (@jwl17330536)
### Breaking Changes
- Removed Google Voice SMS notify platform because the library no longer works.

fabaff committed:
Prepare for next development cycle

balloob committed:
Add support for dynamic frontend panels

balloob committed:
Add panel_iframe component

balloob committed:
Build frontend

balloob committed:
Fix linting errors

balloob committed:
Merge pull request #2537 from home-assistant/frontend-panels

Frontend panels

Commit:
rfxtrx core tests

Commit:
rfxtrx sensor tests

Commit:
rfxtrx switch tests

Commit:
rfxtrx light tests

Danielhiversen committed:
Merge pull request #2539 from home-assistant/rfxtrx_tests

Rfxtrx tests

deisi committed:
repaired dependency of the acer projector switch

Commit:
run gen_requirements.py

fabaff committed:
Merge pull request #2545 from deisi/acer_pyserial_update

repaired dependency of the acer projector switch

especially-relative committed:
Fix bug with imap sensor (#2546)

Fixed bug where the new connection was not saved when a reconnect
attempt was made; broadended the exception catching.

fabaff committed:
Upgrade python-telegram-bot to 5.0.0 (#2542)

fabaff committed:
Upgrade python-twitch to 1.3.0 (#2540)

fabaff committed:
Upgrade cherrypy to 6.1.1 (#2538)

fabaff committed:
Upgrade pytz to 2016.6.1 (#2541)

Commit:
Fix initialization state for GPIO switches configured with inverted logic (#2550)

When switches are configured to use inverted logic, the GPIO pins initial
state has to be inverted as well (set to HIGH)

balloob committed:
Add ensure config script (#2548)

balloob committed:
Update frontend

turbokongen committed:
zxt 120 has changed in ozw (#2551)

Commit:
Update frontend

balloob committed:
Update frontend

pavoni committed:
Tidy / Refactor Vera (#2569)

* Add power attribute to switch.

* Move device_state_attributes into base class.

* Fix imports following refactor.

* Bump pyvera version - should add contributed support for older (UI5) version dimmers and locks.

* Refactor device lookup to be based on vera classes, push category back into library.

* Add generic power attribute, fix inherited class order bug.

* Tidy.

nkgilley committed:
expect a list of devices from config (#2567)

support multiple components. (#2565)

Danielhiversen committed:
Rfxtrx sensor (#2563)

* fire event rfxtrx sensor

* Add fire_event to rfxtrx sensor config

* Add test for rfxtrx fire event in sensor

PetitCircuitLab committed:
Added support for luminance value (#2562)

Danielhiversen committed:
MagicLight/Flux WiFi Color LED Light Component (#2534)

* Initial version for flux light

* Update version of flux_led library

* update flux led

Commit:
Move location lookup before zone checks. (#2557)

n8henrie committed:
Don't overwrite the config directory (#2570)

Closes #2566

The `else` seems to have been an error and was overwriting a non-default config directory with the default location.

americanwookie committed:
Fix services registration, and adding schema util to sonos (#2558)

* Moving service registration into def so that it can be called for both discovery methods

* Adding use of schemas to sonos

balloob committed:
Update frontend

turbokongen committed:
Added more binary sensor and switch classes. Ref.Pepper1 database (#2573)

Commit:
Wink water leak sensor (#2572)

Commit:
Allow templates for delays in scripts (#2560)

robbiet480 committed:
Change path to favicon in GNTP

This broke when #2537 was merged.

fabianhjr committed:
Type Hints - __main__ (#2574)

* Add __main__ type hints

* Fix most errors of __main__

* Add ignore for script.run()

* Add type annotations for from_config_dict and from_config_file

* Fix errors

* Fix requirement error

* Add mypy type check to tests

* Enable travis typing check

* Messed up the tox deps

* Laxer type checker

balloob committed:
Update service worker

turbokongen committed:
Thermostat and hvac status fix for #2465 (#2585)

turbokongen committed:
Missing Fortrezz siren fix for #2581 (#2586)

turbokongen committed:
Positioning issue for zwave rollershutter. fix for #2581 (#2587)

This fixes issue: #2486

balloob committed:
Update frontend

pvizeli committed:
Update pyhomematic to version 0.1.10 (#2589)

balloob committed:
Speed up MyPy test (#2584)

turbokongen committed:
Position fix, updating fix and start-stop for zwave rollershutter (#2594)

fabaff committed:
Upgrade voluptuous to 0.9.1 (#2602)

fabaff committed:
Upgrade netdisco to 0.7.0 (#2598)

fabaff committed:
Upgrade slacker to 0.9.24 (#2597)

fabaff committed:
Add timestamp filters (#2596)

Commit:
Fixed REST API URL in readme.

fabaff committed:
Merge pull request #2605 from rostved/readme-api-url-fix

Fixed REST API URL in readme.

ngraziano committed:
Float value for input slider (#2607)

* Allow input_slider value to be a float number.

* Change input_slider unit test to allow float number.

balloob committed:
Update frontend

fabaff committed:
Upgrade sendgrid to 3.0.7 (#2604)

laf committed:
Added Russound RNET support (#2591)

* Added Russound RNET support

* Fixed farcy issues

* Updated volume_level + fixed requirements_all.txt

* Updated syntax + changed variable

fabianhjr committed:
Type Hints - Core/Utils/Helpers Part 1 (#2592)

* Fix deprecated(moved) import

* Add util/dt typing

* Green on mypy util/dt

* Fix some errors

* First part of yping util/yaml

* Add more typing to util/yaml

kellerza committed:
Update recorder. (#2549)

* Update recorder.

models.py:
 - Use scoped_session in models.py to fix shutdown error
__init__.py:
 - Session _commit & retry method
 - Single session var for purge_data
 - Ensure single _INSTANCE
 - repeat purge every 2 days
 - show correct time in log_error

* _commit

* Restore models to old functionality, swap purge, remove _INSTANCE cleanup from tests, typing ignore Base class

* pylint

* Remove recorder from model unit test

Commit:
Knx thermostat (#2575)

* Major rewrite of the KNX multi address device. This class wasn't used before, but the new class will be the base for the LNX thermostat module

* newer KNXIP version needed as the previous version had a serious bug

* Update knxip to later version

* Added thermostat module

* First implementation of a KNX thermostat module

* Minor cleanup

* Removed unsed code

kellerza committed:
Update Qwikswitch: fix typing, add validation, shutdown (#2603)

* Update Qwikswitch: fix typing, add validation, shutdown

* Delay startup listener, fix validation

* Fix workerpool errors

balloob committed:
Exclude tests in dependencies in test dir from pytest (#2618)

vladonemo committed:
Implementing set_hvac_mode for Nest (#2621)

balloob committed:
Fix test to test Norway fix (#2626)

n8henrie committed:
Add timeout kwarg to call_service() and API.__call__() (#2612)

Fixes #2611

Adds a timeout kwarg to call_service and API.__call__ with default set
to 5 (as per previous behavior). Will not change existing behavior but
will allow remote Python API calls to specify a longer (or shorter)
timeout if they know that a script takes longer than 5 seconds to
return.

balloob committed:
Allow circular dependency with discovery (#2616)

Commit:
Add the option to add additional tags when logging to InfluxDB (#2613)

cbulock committed:
DirecTV Receiver Media Player Component (#2559)

* DirecTV receiver component

* styling cleanup

* Updated coveragerc and requirements all

* using string format

* linter fixes

turbokongen committed:
Make sure zwave values are updated regardles of manual or frontend update, (#2595)

* Make sure values are updated regardles of manual or frontend update,

* Devices with set_switch command was not happy with fast updating.

* Binary triggersensors command was not happy with refreshed updating.

fabaff committed:
Add get_config (#2627)

kellerza committed:
Update icloud device_tracker (#2614)

*  slugify() for dev_id (fixes #2162) [Keep space replacement to not impact known_devices.yaml]
*  pyicloud upgrade 0.9.1
*  config validation
*  Only poll icloud every 4 minutes...
*  Immediately pull device state on HASS start
*  Added new test with icloud char e' acute [chr(233)]
* Suppress pyicloud logging

fabianhjr committed:
Improvement typing core (#2624)

* Add package typing

* Add util/location typing

* FIX: lint wrong order of imports

* Fix sometyping and add helpers/entity typing

* Mypy import trick

* Add asteroid to test requiremts to fix pylint issue

* Fix deprecated function isSet for is_set

* Add loader.py typing

* Improve typing bootstrap

Commit:
Removed Google Voice SMS notification support (#2628)

kellerza committed:
Update icloud to respect track=false. (#2640)

balloob committed:
Use local timezone for log and history dates (#2622)

* Use local timezone for log and history dates

* home-assistant-js fix

* Submodule updates not included so travis can build

* Separate Date and DateTime http validators

* Include submodule reference

* Update frontend

fotoetienne committed:
Add support for x10 lights (#2637)

* Add support for x10 lights

* X10 linting and add to .coveragerc

schneefux committed:
Hyperion active (#2634)

* Hyperion lets you turn it on and off

* Update hyperion to use recent API functions

The plugin now gets the active color from the server.
Add a configuration option "default_color" to customize the turn_on color.

fabaff committed:
Remove Awesoe Light artefacts

fabaff committed:
Merge pull request #2642 from fabaff/x10

Remove Awesome Light artefacts

balloob committed:
Update frontend

nkgilley committed:
check for error while running speedtest (#2643)

pancho-villa committed:
Update to group component to properly handle zone changes in tracked devices (#2631)

* pep8 fixes for group and test

* update to pass linting

* docstring fix.

* reduced length of docstring on test.

americanwookie committed:
Adding tests for sonos registration (#2633)

abcminiuser committed:
Add MPC-HC Media Player Component (#2635)

* Initial media_player component for the MPC-HC web API.

* Update .coveragerc to exclude the MPC-HC media player component.

* We don't need a session for every HTTP fetch.

* Use host in configuration YAML to match Kodi component.

* Fix PyLint errors.

* Fix PEP8 errors and use more idiomatic Python to get dict() values.

* Add MPC-HC remote command capabilities for basic control.

nkgilley committed:
load the last good state from db if speedtest data is None (#2645)

* load the last good state from db if speedtest data is None

* return if recorder is not available

balloob committed:
Add custom panel example using React (#2651)

pvizeli committed:
fix telegram bug (#2653)

balloob committed:
Update panel.html

Commit:
Add group state for locks (#2647)

* Add group state for locks

Added  ", (STATE_LOCKED, STATE_UNLOCKED)" to _GROUP_TYPES

Don't have a working HA right now, so can't test..

* Modified from homeassistant.const import

* Removed white space

* Line length change

* Removed white space.. again!

Commit:
Merge branch 'master' into dev

fabaff commented:
Release notes: https://github.com/home-assistant/home-assistant.io/blob/next/source/_posts/2016-07-30-tbd.markdown

balloob commented:
@americanwookie you managed to sneak in some PEP257 failures 😦

mezz64 committed:
Expand to respond to basic node events (#2615)

Allows zwave devices that can only push out basic set commands to be
captured by hass as zwave.node_events.

americanwookie committed:
Fixing PEP257 issues in #2633 (#2658)

nkgilley committed:
Install mysqlclient and psycopg2 (#2662)

I don't know if this is the right place for this, but I'm tired of having to install mysqlclient or psycopg2 after every docker update if I want to use mysql of postgres.

fabaff committed:
Add timeout (fixes #2661) (#2666)

balloob committed:
Add commented out default password (#2656)

balloob committed:
Update frontend

laf commented:
Awesome work @balloob 👍
